tests/e2e/main.go

```diff
@@ -0,0 +1,433 @@
+	cmd := exec.Command(binaryPath, args...)
+	cmd.Stdin = os.Stdin
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	cmd.Dir = assetsDir
+	return cmd.Run()
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "e2e",
+	Short: "Test deployment of cluster-api stack",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		kubeconfig := cmd.Flag("kubeconfig").Value.String()
+		logLevel := cmd.Flag("log-level").Value.String()
+		maoImage := cmd.Flag("mao-image").Value.String()
+		assetsPath := cmd.Flag("assets-path").Value.String()
+		clusterID := cmd.Flag("cluster-id").Value.String()
+
+		if kubeconfig == "" {
+			return fmt.Errorf("--kubeconfig option is required")
+		}
+
+		log.SetOutput(os.Stdout)
+		if lvl, err := log.ParseLevel(logLevel); err != nil {
+			log.Panic(err)
+		} else {
+			log.SetLevel(lvl)
+		}
+
+		testConfig := NewTestConfig(kubeconfig)
+
+		// create terraform stub enviroment
+		if err := cmdRun(assetsPath, "terraform", "init"); err != nil {
+			glog.Fatalf("unable to run terraform init: %v", err)
+		}
+		tfVars := fmt.Sprintf("-var=enviroment_id=%s", clusterID)
+		if err := cmdRun(assetsPath, "terraform", "apply", tfVars, "-auto-approve"); err != nil {
+			glog.Fatalf("unable to run terraform apply -auto-approve: %v", err)
+		}
+		defer tearDown(testConfig, assetsPath)
+
+		// create mao deployment and assumptions, i.e secrets, namespaces, appVersion, mao config
+		// generate aws creds kube secret
+		if err := cmdRun(assetsPath, "./generate.sh"); err != nil {
+			glog.Fatalf("unable to run generate.sh: %v", err)
+		}
+
+		// generate assumed namespaces
+		if err := createNamespace(testConfig, targetNamespace); err != nil {
+			return err
+		}
+
+		// create status CRD
+		apiextensionsscheme.AddToScheme(scheme.Scheme)
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		if statusCRD, err := ioutil.ReadFile(filepath.Join(assetsPath, "manifests/status-crd.yaml")); err != nil {
+			glog.Fatalf("Error reading %#v", err)
+		} else {
+			CRDObj, _, err := decode([]byte(statusCRD), nil, nil)
+			if err != nil {
+				glog.Fatalf("Error decoding %#v", err)
+			}
+			CRD := CRDObj.(*apiextensionsv1beta1.CustomResourceDefinition)
+
+			if err := createCRD(testConfig, CRD); err != nil {
+				return err
+			}
+		}
+
+		// genereate mao config
+		maoConfigTemplateData, err := ioutil.ReadFile(filepath.Join(assetsPath, "manifests/mao-config.yaml"))
+		if err != nil {
+			glog.Fatalf("Error reading %#v", err)
+		}
+		configValues := &render.OperatorConfig{
+			AWS: &render.AWSConfig{
+				ClusterID:   clusterID,
+				ClusterName: clusterID,
+				Region:      region,
+			},
+		}
+		maoConfigPopulatedData, err := render.Manifests(configValues, maoConfigTemplateData)
+		if err != nil {
+			glog.Fatalf("Unable to render manifests %q: %v", maoConfigTemplateData, err)
+		} else {
+			mapConfigMap := &apiv1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "cluster-config-v1",
+					Namespace: "kube-system",
```

Comment:
should we revert namespace [here](https://github.com/openshift/machine-api-operator/blob/cvo-integration/cmd/main.go#L12) to `kube-system` or have have mao specific namespace here as well?